server/tilesource/base.py

```diff
@@ -352,29 +576,218 @@ def getRegion(self, width=None, height=None, **kwargs):
+            'ceil' or 'round', that function is used to convert the level to an
+            integer (the exact flag still applies).  If None, the level is not
+            cropped to the actual image's level range.
+        :returns: the selected level or None for no match.
+        """
+        mag = self.getMagnificationForLevel()
+        ratios = []
+        if magnification and mag['magnification']:
+            ratios.append(float(magnification) / mag['magnification'])
+        if mm_x and mag['mm_x']:
+            ratios.append(mag['mm_x'] / mm_x)
+        if mm_y and mag['mm_y']:
+            ratios.append(mag['mm_y'] / mm_y)
+        ratios = [math.log(ratio) / math.log(2) for ratio in ratios]
+        # Perform some slight rounding to handle numerical precision issues
+        ratios = [round(ratio, 4) for ratio in ratios]
+        if not len(ratios):
+            return None if exact else mag['level']
+        if exact:
+            if any(int(ratio) != ratio or ratio != ratios[0]
+                   for ratio in ratios):
+                return None
+        ratio = round(sum(ratios) / len(ratios), 4)
+        level = mag['level'] + ratio
+        if rounding:
+            level = int(math.ceil(level) if rounding == 'ceil' else
+                        round(level))
+        if (exact and (level > mag['level'] or level < 0) or
+                (rounding == 'ceil' and level > mag['level'])):
+            return None
+        if rounding is not None:
+            level = max(0, min(mag['level'], level))
+        return level
+
+    def tileIterator(self, format=(TILE_FORMAT_PIL, ), resample=False,
+                     **kwargs):
+        """
+        Iterate on all tiles in the specifed region at the specified scale.
+        Each tile is returned as part of a dictionary that includes
+            x, y: (left, top) coordinates in current magnification pixels
+            width, height: size of current tile in pixels
+            tile: cropped tile image
+            format: format of the tile
+            level: level of the current tile. None if not present
+            level_x, level_y: the tile reference number within the level.
+            magnification: magnification of the current tile
+            mm_x, mm_y: size of the current tile pixel in millimeters.
+            gx, gy: (left, top) coordinate in maximum-resolution pixels
+            gwidth, gheight: size of of the current tile in maximum resolution
+                pixels.
+        If a region that includes partial tiles is requested, those tiles are
+        cropped appropriately.  Most images will have tiles that get cropped
+        along the right and bottom egdes in any case.  If an exact
+        magnification or scale is requested, no tiles will be returned unless
+        allowInterpolation is true.
+
+        :param format: the desired format or a tuple of allowed formats.
+            Formats are members of TILE_FORMAT_*.
```

Comment:
Will probably be helpful if you list all formats here with a brief description or at least put a pointer to the place where they are described. Because this is what the developers will most-likely look at when they want to create an iterator.